pandas/_libs/tslibs/period.pyx

```diff
@@ -163,6 +111,31 @@ cdef extern from *:
     int FR_UND
 
 
+cdef int max_value(int left, int right) nogil:
+    if left > right:
+        return left
+    return right
+
+
+cdef int min_value(int left, int right) nogil:
+    if left < right:
```

Comment:
grr how many times do we have this i the code base :)

Reply:
We could probably use np_datetime.cmp_scalar here, would need to cast from int to int64_t